### ISSUE TITLE
Missing reflective registration (e.g. CopySource)

### ISSUE THREAD

**Issue by JiriOndrusek** [quarkiverse/quarkiverse-minio#21]
I was implementing camel-quarkus minio extension by using quarkiverse-minio (https://github.com/apache/camel-quarkus/pull/2116).
I've found that there is some missing registration for reflaction. For example there is a missing constructor for [CopySource](https://github.com/minio/minio-java/blob/master/api/src/main/java/io/minio/CopySource.java) which is called from its builder.

My solution used in camel-quarkus minio extension is:
```
    @BuildStep
    ReflectiveClassBuildItem registerForReflection(CombinedIndexBuildItem combinedIndex) {
        IndexView index = combinedIndex.getIndex();
        String[] dtos = index.getAllKnownSubclasses(DotName.createSimple(BaseArgs.class.getName())).stream()
                .map(c -> c.name().toString()).toArray(String[]::new);
        return new ReflectiveClassBuildItem(false, false, dtos);
    }
```

**Comment by jtama-op** [comment]
Ok I will check. 
For understanding, the solution you used is the one you injected in your extension to solve the issue ?

**Comment by JiriOndrusek** [comment]
Yes.

**Comment by jtama-op** [comment]
Released in 0.2.0
